### PR TITLE
feat(nuxt): scan and register layouts in nested folders

### LINKED ISSUE #14275: Nuxt 3 does not respect layouts in separate folders
#### Nuxt Version

- nuxt:  v3.0.0 rc-4

#### Steps to reproduce

On a clean Nuxt 3 install.

Create a layout inside a folder like.

`layouts/mobile/custom.vue`

Use it in a page as follows.

`
<NuxtLayout name="mobile/custom">
    some code ...
</NuxtLayout>
`

### Describe the bug

When I create a layout inside a folder like "layouts/mobile/custom.vue" and use it inside of "pages/index.vue" on a clean install of Nuxt 3 as follows.

`
<NuxtLayout name="mobile/custom">
    some code ...
</NuxtLayout>
`

It does not work, I don't know if this feature is not supported yet or if it is a bug, because when I went through the Nuxt 2 info I found this which came out thanks to an issue of this exact same problem in Nuxt 2

[Adds support for folders in /layouts nuxt/nuxt.js#12765](https://github.com/nuxt/nuxt.js/pull/1865)

Normal layout use like Nuxt 3 documentation works just fine, "layouts/custom.vue" inside of "pages/index.vue" like this.

`
<NuxtLayout name="custom">
    some code ...
</NuxtLayout>
`

### What is Expected?

Is expected to work exactly as normal layouts would.

### What is actually happening?

Got an error in console as if the layout does not exists.

`Invalid layout "mobile/custom" selected.`

### PR DESCRIPTION
**Pull request by Suv4o**
fix(nuxt): Nuxt 3 does not respect layouts in separate folders (#14275)

### 🔗 Linked issue

resolves #14275

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Fix the issue of not being able to create layouts in different directories, as specified in Issue #14275.

Now, we can define layouts in the following format:

<p dir="auto" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 16px; color: rgb(230, 237, 243); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(13, 17, 23); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">For example:</p>

File | Layout name
-- | --
~/layouts/desktop/default.vue | 'desktop-default'
~/layouts/desktop-base/base.vue | 'desktop-base'
~/layouts/desktop/index.vue | 'desktop'

```
<template>
    <NuxtLayout name="desktop-base">
      <NuxtPage />
    </NuxtLayout>
</template>
```

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by danielroe — DISMISSED** [review]
Wonderful - this looks great! ❤️ 

Just one change - rather than placing the normalisation logic here in the runtime, can we do it in the build time context (`packages/nuxt/src/core/app.ts`)?

**Comment by daktas** [comment]
@pi0 / @danielroe / @Atinux 
anybody to validate this usefull PR ?

**Comment by danielroe** [comment]
I have already reviewed and requested a change.

File: packages/nuxt/src/core/utils/names.ts [hunk 1]
```diff
@@ -1,8 +1,12 @@
...
 import { kebabCase } from 'scule'
+import { withTrailingSlash } from 'ufo'
 
-export function getNameFromPath (path: string) {
-  return kebabCase(basename(path).replace(extname(path), '')).replace(/["']/g, '')
+export function getNameFromPath (path: string, relativeTo?: string) {
+  const relativePath = relativeTo
+    ? normalize(path).replace(withTrailingSlash(normalize(relativeTo)), '')
+    : basename(path)
+  return kebabCase(relativePath.replace(/[\\/]+/g, '-').replace(/\/index\.\w+$/, '').replace(extname(relativePath), '')).replace(/["']/g, '')
```
Comment (danielroe) [inline packages/nuxt/src/core/utils/names.ts:9]:
```suggestion
  return kebabCase(relativePath.replace(/\/index\.\w+$/, '').replace(/[\\/]+/g, '-').replace(extname(relativePath), '')).replace(/["']/g, '')
```

**Comment by Atinux** [comment]
Looks good to me, would be nice to update the documentation about it (can be done later)

**Review by danielroe — APPROVED** [review]

**Pull request merged**